test/powershell/Modules/Microsoft.PowerShell.Utility/Update-FormatData.Tests.ps1

```diff
@@ -48,12 +48,60 @@ Describe "Update-FormatData basic functionality" -Tags "CI" {
                     </Type>
                 </Types>
 "@
-		$xmlContent > $testfile
+        $xmlContent > $testfile
     }
 
-	It "Update-FormatData with WhatIf should work"{
+    It "Update-FormatData with WhatIf should work"{
 
         { Update-FormatData -Append $testfile -WhatIf } | Should Not Throw
         { Update-FormatData -Prepend $testfile -WhatIf } | Should Not Throw
-	}
+    }
+}
+
+
+Describe "Update-FormatData with resources in CustomControls" -Tags "CI" {
+
+    BeforeAll {
+        $templatePath = Join-Path $PSScriptRoot (Join-Path 'assets' 'UpdateFormatDataTests.format.ps1xml')
+        $formatFilePath = Join-Path $TestDrive 'UpdateFormatDataTests.format.ps1xml'
+        $ps = [powershell]::Create()
+        $iss = [system.management.automation.runspaces.initialsessionstate]::CreateDefault2()
+        $rs = [system.management.automation.runspaces.runspacefactory]::CreateRunspace($iss)
+        $rs.Open()
+        $ps.Runspace = $rs
+    }
+    AfterAll {
+        $rs.Close()
+        $ps.Dispose()
+    }
+    Context "Validate Update-FormatData" {
+        It "Resources in WindowsPS syntax should be loaded successfully" {
+            $format = Get-Content -Path $templatePath -Raw
+            $format.Replace("%BaseName%","FileSystemProviderStrings") | Set-Content -Path $formatFilePath -Force
+            $null = $ps.AddScript("Update-FormatData -PrependPath $formatFilePath")
+            $ps.Streams.Error.Clear()
+            $ps.Invoke()
+            $ps.HadErrors | Should be $false

```

Comment:
Would testing `$ps.Streams.Error` give a more meaningful error if something went wrong in the test?